main/tasks/power_management_task.c

```diff
@@ -115,7 +100,11 @@ void POWER_MANAGEMENT_task(void * pvParameters)
 
         if (nvs_config_get_u16(NVS_CONFIG_AUTO_FAN_SPEED, 1) == 1) {
 
-            power_management->fan_perc = (float)automatic_fan_speed(power_management->chip_temp_avg, GLOBAL_STATE);
+            pid_input = power_management->chip_temp_avg;
+            pid_compute(&pid);
+            power_management->fan_perc = (uint16_t) pid_output;
+            Thermal_set_fan_percent(GLOBAL_STATE->device_model, pid_output / 100.0);
+            ESP_LOGI(TAG, "Temp: %.1f°C, SetPoint: %.1f°C, Output: %.1f%%", pid_input, pid_setPoint, pid_output);
```

Comment:
Also log `power_management->fan_rpm`. This will be useful when troubleshooting weird fan setups.